util/src/test/java/care/smith/fts/util/fhir/FhirTagTest.java

```diff
@@ -0,0 +1,193 @@
+package care.smith.fts.util.fhir;
+
+import static org.assertj.core.api.Assertions.*;
+
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Meta;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Patient;
+import org.junit.jupiter.api.Test;
+
+public class FhirTagTest {
+
+  private static final String SYSTEM = "http://smith.care/fhir/tags";
+  private static final String CODE = "test-tag";
+  private static final String DISPLAY = "Test Tag";
+
+  @Test
+  public void testNullBundle() {
+    assertThat(FhirTag.addTagToAllResources(null, SYSTEM, CODE, DISPLAY)).isNull();
+  }
+
+  @Test
+  public void testEmptyBundle() {
+    var emptyBundle = new Bundle();
+    var result = FhirTag.addTagToAllResources(emptyBundle, SYSTEM, CODE, DISPLAY);
+
+    assertThat(result).isNotNull();
+    assertThat(result.getMeta()).isNotNull();
+    assertThat(result.getMeta().getTag()).hasSize(1);
+    assertThat(result.getMeta().getTag().getFirst().getSystem()).isEqualTo(SYSTEM);
+    assertThat(result.getMeta().getTag().getFirst().getCode()).isEqualTo(CODE);
+    assertThat(result.getMeta().getTag().getFirst().getDisplay()).isEqualTo(DISPLAY);
```

Comment:
On lines 61-66 you use satisfies after asserting the size, can we utilize this here, too?